src/providers/core/mod.rs

```diff
@@ -263,13 +262,28 @@ impl ProviderBuilder {
             provider_info_vec.push(provider_info);
         }
 
-        let crate_version: Version = Version::from_str(version!()).map_err(|e| {
-            format_error!("Error parsing the crate version", e);
+        let crate_version: std::result::Result<Vec<u32>, ParseIntError> = env!("CARGO_PKG_VERSION")
+            .split('.')
+            .map(|v| v.parse())
+            .collect();
+
+        let crate_version = crate_version.map_err(|_| {
```

Comment:
Can you also include `format_error` and print out the error message?

Reply:
Done!